src/main.rs

```diff
@@ -95,59 +125,83 @@ fn model_check(args: ArgMatches) -> Result<(), Box<dyn Error>> {
         ATLVertex::FULL { state: 0, formula },
         num_cpus::get() as u64,
     );
-    println!("{:?}", result);
 
-    Ok(())
+    match args.value_of("output") {
+        Some(path) => {
+            let mut file = File::create(path)?;
+            let mut writer = BufWriter::new(&file);
+            write!(&mut writer, "Result: {:?}", result);
+            Ok(())
+        }
+        _ => {
+            println!("Result: {:?}", result);
+            Ok(())
+        }
+    }
 }
 
 /// Define and parse command line arguments
 fn parse() -> ArgMatches<'static> {
+    fn build_common_arguments<'a>(builder: clap::App<'a, 'a>) -> App<'a, 'a> {
+        builder.arg(
+            Arg::with_name("input_model")
+                .short("m")
+                .long("model")
+                .env("INPUT_MODEL")
+                .required(true)
+                .help("The input file to generate model from"),
+        )
+            .arg(
+                Arg::with_name("model_type")
+                    .short("t")
+                    .long("model-type")
+                    .env("MODEL_TYPE")
+                    .required(true)
+                    .help("The type of input file given {{lcgs, json}}"),
+            )
+            .arg(
+                Arg::with_name("formula")
+                    .short("f")
+                    .long("formula")
+                    .env("FORMULA")
+                    .required(true)
+                    .help("The formula to check for"),
+            )
+            .arg(
+                Arg::with_name("output")
+                    .short("o")
+                    .long("output")
+                    .env("OUTPUT")
+                    .help("The path to write output to"),
+            )
+    }
+
+    let mut parser = App::new(PKG_NAME)
+        .version(VERSION)
+        .author(AUTHORS);
+    parser = build_common_arguments(parser);
```

Comment:
This seems to be unused. Is that correct?

Reply:
Oops, old testing code.

Reply:
Removed in af880a9